crmsh/bootstrap.py

```diff
@@ -2136,6 +2139,68 @@ def remove_node_from_cluster(node):
     invoke("corosync-cfgtool -R")
 
 
+def ssh_stage_finished():
+    """
+    Dectect if the ssh stage is finished
+    """
+    key_file_manager = ssh_key.KeyFileManager(sh.cluster_shell())
+    keys = key_file_manager.list_public_key_for_user(None, 'hacluster')
+    return bool(keys)
+
+
+def csync2_stage_finished():
+    """
+    Dectect if the csync2 stage is finished
+    """
+    return ServiceManager().service_is_active(CSYNC2_SERVICE)
+
+
+def corosync_stage_finished():
+    """
+    Dectect if the corosync stage is finished
+    """
+    try:
+        conf_parser.ConfParser.verify_config_file()
+    except ValueError as e:
+        logger.error(e)
+        return False
+    return True
+
+
+INIT_STAGES_DEPENDENCY_MAP = {
```

Comment:
This is a bad naming. It sounds like "ssh" stage depends on `ssh_stage_finished`.

It will be easier to understand if divided into 2 structures:   `INIT_STAGE_CHECKER = {"ssh": ssh_stage_finished, ...}` and `INIT_STAGE_ORDER = ["ssh", "csync2", ...]`